app/src/Services/auth/AuthProvider.test.tsx

```diff
@@ -0,0 +1,272 @@
+      tokens: { accessToken: 'foobar' },
+    })
+    const { result } = renderHook(() => useAuth(), {
+      wrapper: wrapper(mockAuthClient),
+    })
+    expect(result.current.loading).toBe(true)
+
+    await waitFor(() => {
+      expect(result.current.loading).toBe(false)
+      expect(result.current.authenticated).toBe(false)
+    })
+
+    mockAuthClient.getCurrentUser.mockResolvedValue({
+      email: 'foo@bar.com',
+      sub: '123',
+    })
+
+    act(() => {
+      result.current.login('foo@bar.com')
+    })
+    expect(result.current.loading).toBe(true)
+
+    await waitFor(() => {
+      expect(result.current.loading).toBe(false)
+      expect(result.current.authenticated).toBe(true)
+    })
+  })
+
+  test('initialises with authenticated state correctly', async () => {
+    mockAuthClient.getIdToken.mockResolvedValue('id-token')
+    mockAuthClient.getCurrentUser.mockResolvedValue({
+      email: 'foo@bar.com',
+      sub: '123',
+    })
+    const { result } = renderHook(() => useAuth(), {
+      wrapper: wrapper(mockAuthClient),
+    })
+
+    expect(result.current.loading).toBe(true)
+    await waitFor(() => {
+      expect(result.current.loading).toBe(false)
+    })
+
+    expect(result.current.authenticated).toBe(true)
+    expect(result.current.user).toEqual({
+      email: 'foo@bar.com',
+      sub: '123',
+    })
+  })
+
+  test('does not perform multiple authentication attempts', async () => {
+    mockAuthClient.authorize.mockResolvedValue({
+      tokens: { accessToken: 'foobar' },
+    })
+    mockAuthClient.getCurrentUser.mockResolvedValue({
+      email: 'foo@bar.com',
+      sub: '123',
+    })
+    const { result } = renderHook(() => useAuth(), {
+      wrapper: wrapper(mockAuthClient),
+    })
+
+    await waitFor(() => expect(result.current.loading).toBe(false))
+    act(() => {
+      result.current.login('email')
+    })
+    expect(result.current.loading).toBe(true)
+    act(() => {
+      result.current.login('email')
+    })
+    await waitFor(() => expect(result.current.loading).toBe(false))
+    expect(result.current.authenticated).toBe(true)
+
+    expect(mockAuthClient.authorize).toHaveBeenCalledTimes(1)
+  })
+
+  describe('blockUntilInitialised', () => {
+    test('only renders children after initialisation', async () => {
+      mockAuthClient.getCurrentUser.mockResolvedValue(undefined)
+      const blockingWrapper = ({ children }: { children: ReactNode }) => (
+        <AuthProvider
+          client={mockAuthClient as any}
+          blockUntilInitialised={true}
+        >
+          {children}
+        </AuthProvider>
+      )
+      const { queryByText } = render(<Text>Authentication test</Text>, {
+        wrapper: blockingWrapper,
+      })
+
+      expect(queryByText('Authentication test')).toBeNull()
+      await waitFor(() =>
+        expect(queryByText('Authentication test')).toBeTruthy(),
+      )
+    }, 10000) // Without this, the test fails on GitHub Actions run.
```

Comment:
I don't know why this is, the test covers a simple `useEffect` hook with an async function that should resolve immediately on next tick. It passes locally without issue, and I have used this technique before in other codebases with any CI issues. I'm comfortable leaving it in place for now until someone has time to properly investigate.